src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.cs

```diff
@@ -94,11 +94,12 @@ public static partial class StreamPipeExtensions
     }
     public partial class StreamPipeReaderOptions
     {
-        public StreamPipeReaderOptions(System.Buffers.MemoryPool<byte>? pool = null, int bufferSize = -1, int minimumReadSize = -1, bool leaveOpen = false) { }
+        public StreamPipeReaderOptions(System.Buffers.MemoryPool<byte>? pool = null, int bufferSize = -1, int minimumReadSize = -1, bool leaveOpen = false, bool waitForData = false) { }
```

Comment:
You have to add new constructor. Adding new default arg to existing constructor is binary breaking change.

Reply:
OK changed it